src/libostree/ostree-repo-commit.c

```diff
@@ -2377,6 +2409,9 @@ ostree_repo_commit_transaction (OstreeRepo                  *self,
   if (!rename_pending_loose_objects (self, cancellable, error))
     return FALSE;
 
+  if (!fsync_object_dirs (self, cancellable, error))
+    return FALSE;
```

Comment:
Is there a purpose to the refactor into a separate function? Seems like `fsync` behaviour in this path hasn't really changed here, right?

Reply:
In `per-object-fsync` mode we aren't using the staging dir; we still allocate one because making that conditional would be a notably bigger patch.  So we need to sync the actual target `objects/` directory.  And in any case I think we can get away with doing it this way rather than syncing the staging dir in both cases.